backend/services/questions.py

```diff
@@ -0,0 +1,72 @@
+"""Service to handle password generation and management"""
+
+import os
+from typing import List
+
+from ..models import QuestionsPublic, Question, Document
+
+__authors__ = ["Nicholas Almy"]
+
+
+class QuestionService:
+    """A service to handle questions for the competition"""
+
+    def __init__(self):
+        # Walk through the questions directory and load all the questions
+        self.questions = self.load_questions()
```

Comment:
In my recent changes, I decided to make this a class variable so that we can still force a reload if we need to. I also made the code less monolithic and adhered to better practice, so could you take another look ? 🥺

Reply:
Okay I was finally able to sit down and do some testing. It looks like FastAPI constructs a new instance of this class every time that a different user runs the route. So this means that we will generate a new instance of `QuestionsPublic` each time that a new user runs the `/api/questions/` route. I don't think this is the behavior we want here. A possible fix that I just tried is to make all the methods in the `QuestionService` class methods then run the `load_questions()` in the api file like this:

![image](https://github.com/user-attachments/assets/3ab42fa0-04c3-44f7-964e-af2caac03438)

I'm not sure how much this adheres to "good practice" but it works.